packages/cli-hooks/src/start.js

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env node
+
+import { fileURLToPath } from 'url';
+import childProcess from 'child_process';
+import path from 'path';
+import fs from 'fs';
+
+/**
+ * Implementation of the start hook that begins a new process to run the app.
+ * The child processes will continue until exited or interrupted by the caller.
+ */
+
+if (fs.realpathSync(process.argv[1]) === fileURLToPath(import.meta.url)) {
+  start(process.cwd());
+}
+
+/**
+ * Start hook implementation that verifies and runs an app in Socket Mode.
+ * @param {string} cwd - The current working directory of the project.
+ */
+export default function start(cwd) {
+  validateEnvironment();
+
+  const customPath = process.env.SLACK_CLI_CUSTOM_FILE_PATH;
+  const pkgJSONMain = getPackageJSONMain(cwd);
+  const pkgJSONDefault = 'app.js';
+  const fullPath = path.resolve(cwd, customPath || pkgJSONMain || pkgJSONDefault);
+
+  const app = childProcess.spawn('node', [`${fullPath}`]);
+  app.stdout.setEncoding('utf-8');
+  app.stdout.on('data', (data) => {
+    process.stdout.write(data);
+  });
+  app.stderr.on('data', (data) => {
+    process.stderr.write(data);
+  });
+  app.on('close', (code) => {
+    console.log(`Local run exited with code ${code}`); // eslint-disable-line no-console
+  });
+}
+
+/**
+ * Gathers the main value from the package.json for the project if present.
+ * @param {string} cwd - The current working directory of the project.
+ * @returns {string | undefined} - The main script to run for the project.
+ */
+function getPackageJSONMain(cwd) {
+  try {
+    const packageJSONPath = path.join(cwd, 'package.json');
+    const { main } = JSON.parse(fs.readFileSync(packageJSONPath, 'utf-8'));
+    return main;
+  } catch {
+    return undefined;
+  }
+}
+
+/**
+ * Confirms environment variables are prepared by the CLI.
+ */
+function validateEnvironment() {
+  if (!process.env.SLACK_CLI_XOXB) {
+    throw new Error('Missing local run bot token. Please see Slack CLI maintainers to troubleshoot.');
+  }
+  if (!process.env.SLACK_CLI_XAPP) {
+    throw new Error('Missing local run app token. Please see Slack CLI maintainers to troubleshoot');
```

Comment:
These are probably not the instructions we should provide to devs hitting these errors... not sure what else to suggest though 🤔

Reply:
This might be a bit verbose, but I think this might be more useful. I still remain open to any suggestions!

```sh
$ node ~/programming/tools/node-slack-sdk/packages/cli-hooks/src/start.js
file:///Users/ethan.zimbelman/programming/tools/node-slack-sdk/packages/cli-hooks/src/start.js:66
    throw new Error(missingBotTokenError);
          ^

Error: Missing the bot token needed to start the app with Socket Mode.
Hints: Setting the SLACK_CLI_XOXB environment variable is required.
Check: Confirm that you are using the latest version of the Slack CLI.
    at validateEnvironment (file:///Users/ethan.zimbelman/programming/tools/node-slack-sdk/packages/cli-hooks/src/start.js:66:11)
    at start (file:///Users/ethan.zimbelman/programming/tools/node-slack-sdk/packages/cli-hooks/src/start.js:22:3)
    at file:///Users/ethan.zimbelman/programming/tools/node-slack-sdk/packages/cli-hooks/src/start.js:14:3
    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:66:12)
```

Reply:
Yeah, awesome, much better. Thanks!